src/components/Header/Header.js

```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import PropTypes from 'prop-types';
+import data from '../../../data';
+import { HeaderSection } from './HeaderSection/HeaderSection';
+
+export const Header = (props) => {
+  const {
+    className,
+    header,
+    ...otherProps
```

Comment:
Skoro nie wykorzystujesz ich w komponencie to nie ma sensu ich destrukturyzować czy "zbierać resztę".

Warto trzymać się zasady YAGNI: https://pl.wikipedia.org/wiki/YAGNI